drivers/usbdev/cdcacm.c

```diff
@@ -3430,8 +3430,53 @@ void cdcacm_uninitialize(FAR struct usbdevclass_driver_s *classdev)
 {
   FAR struct cdcacm_driver_s *drvr = (FAR struct cdcacm_driver_s *)classdev;
   FAR struct cdcacm_dev_s    *priv = drvr->dev;
-  char devname[CDCACM_DEVNAME_SIZE];
+
+  cdcacm_uninitialize_instance(priv->minor, classdev);
+}
+
+/****************************************************************************
+ * Name: cdcacm_uninitialize_instance
+ *
+ * Description:
+ *   Function to uninitialize specific cdcacm instance
+ *
+ * Input Parameters:
+ *   minor - CDCACM node minor number
+ *
+ * Returned Value:
+ *   OK when successful, -ENODEV if cdcacm is not initialized
+ *
+ ****************************************************************************/
+
+int cdcacm_uninitialize_instance(int minor,
+                                 FAR struct usbdevclass_driver_s *classdev)
+{
   int ret;
+  FAR struct cdcacm_driver_s *drvr = (FAR struct cdcacm_driver_s *)classdev;
+  FAR struct cdcacm_dev_s *priv;
+  char devname[CDCACM_DEVNAME_SIZE];
+
+  /* Create device node path from minor number */
+
+  snprintf(devname, sizeof(devname), CDCACM_DEVNAME_FORMAT, minor);
+
+  /* If classdev is not provided, find it from the file system */
+
+  if (!classdev)
+    {
+      FAR struct cdcacm_alloc_s *cdcacm_alloc = find_driver(devname);
+      if (cdcacm_alloc)
+        {
+          drvr = &cdcacm_alloc->drvr;
```

Comment:
should we return inode with refcount like find_blockdriver to avoid other threads destroy the driver to make drv pointer stale.

Reply:
user may unplug cable randomly which may make the pointer invaid.

Reply:
If this is an issue, it is an issue also now, right? This PR doesn't change that behaviour. There is no locking with the pointer atm. if it is stored in "cdcacm" in CONFIG_BUILD_FLAT or CONFIG_BUILD_PROTECTED.

We can add a mutex around cdcacm_intialize, and cdcacm_uninitialize_instance, if you like. But that is not relally related to this PR.

In any case, device protecting itself should not be done with inode lock or inode refcount. Those are only for the filesystem.